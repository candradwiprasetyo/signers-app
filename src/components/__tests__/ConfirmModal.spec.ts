import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ConfirmModal from '@/components/ConfirmModal.vue'

describe('ConfirmModal', () => {
  it('render correctly layout', () => {
    const wrapper = mount(ConfirmModal)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('emit cancel when background is clicked', async () => {
    const wrapper = mount(ConfirmModal)
    await wrapper.trigger('click')
    expect(wrapper.emitted('cancel')).toBeTruthy()
  })

  it('emit cancel when click batal', async () => {
    const wrapper = mount(ConfirmModal)
    const buttons = wrapper.findAll('button')
    const cancelBtn = buttons.find((btn) => btn.text() === 'Batal')
    expect(cancelBtn).toBeTruthy()
    await cancelBtn!.trigger('click')
    expect(wrapper.emitted('cancel')).toBeTruthy()
  })

  it('emit confirm when click hapus', async () => {
    const wrapper = mount(ConfirmModal)
    const buttons = wrapper.findAll('button')
    const confirmBtn = buttons.find((btn) => btn.text() === 'Hapus')
    expect(confirmBtn).toBeTruthy()
    await confirmBtn!.trigger('click')
    expect(wrapper.emitted('confirm')).toBeTruthy()
  })
})
