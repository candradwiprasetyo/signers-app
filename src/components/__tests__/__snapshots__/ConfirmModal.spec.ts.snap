// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfirmModal > render correctly layout 1`] = `
"<div class="fixed inset-0 z-50 bg-black/30 backdrop-blur-sm flex items-center justify-center">
  <div class="bg-white p-6 rounded-xl shadow-xl w-full max-w-sm text-center">
    <h2 class="text-lg font-semibold text-gray-800 mb-4">Konfirmasi Hapus</h2>
    <div class="mb-4 p-4 text-sm text-yellow-800 bg-yellow-100 border border-yellow-300 rounded"> Fitur delete dibatasi oleh server. Tidak dapat mengakses DELETE API. </div>
    <p class="text-gray-600 mb-6">Apakah Anda yakin ingin menghapus data ini?</p>
    <div class="flex justify-end gap-3"><button class="px-4 py-2 text-sm rounded bg-gray-200 hover:bg-gray-300 text-gray-800"> Batal </button><button class="px-4 py-2 text-sm rounded bg-red-500 hover:bg-red-600 text-white"> Hapus </button></div>
  </div>
</div>"
`;
